ocaml/lambda/transl_list_comprehension.ml

```diff
@@ -192,19 +188,22 @@ let iterator ~transl_exp ~scopes = function
   | Texp_comp_in { pattern; sequence } ->
       let iter_list =
         Let_binding.make (Immutable Strict) (Pvalue Pgenval)
-          "iter_list" (transl_exp ~scopes sequence)
+          "iter_list" (transl_exp ~scopes Sort.for_predef_value sequence)
       in
       (* Create a fresh variable to use as the function argument *)
       let element = Ident.create_local "element" in
       { builder      = rev_dlist_concat_map
       ; arg_lets     = [iter_list]
       ; element
       ; element_kind =
-          Typeopt.layout pattern.pat_env pattern.pat_loc pattern.pat_type
+          Typeopt.layout pattern.pat_env pattern.pat_loc
+            Layouts.Sort.for_predef_param pattern.pat_type
```

Comment:
This is something like `Sort.for_list_element`.

Reply:
Won't that always just be `value`, though?

Reply:
Not when we lift the mixed-block restriction.